core/controllers/base_test.py

```diff
@@ -1423,3 +1469,397 @@ def test_that_correct_derived_class_does_not_raise_exception(self):
         with self.swap(self, 'testapp', self.mock_testapp):
             self.post_json(
                 '/correctmock', payload, expected_status_int=200)
+
+
+class SchemaValidationIntegrationTests(test_utils.GenericTestBase):
+    """Tests all the functionality of SVS(Schema-Validation-System)
+    architecture.
+    """
+    handler_class_names_with_no_schema = (
+        payload_validator.HANDLER_CLASS_NAMES_WITH_NO_SCHEMA)
+    wiki_page_link = (
+        'https://github.com/oppia/oppia/wiki/Validation-of-handler-args')
+
+    def _get_list_of_routes_which_need_schemas(self):
+        """This method iterates over all the routes and returns those routes
+        which need schema.
+
+        Returns:
+            list_of_routes_which_needs_schema: list(RedirectRoute). A list of
+            RedirectRoute objects.
+        """
+        list_of_routes_which_needs_schema = []
+        for route in main.URLS:
+            # TODO(#13139): Remove below check once all the MAPREDUCE_HANDLERS
+            # are removed from the codebase.
+            if isinstance(route, tuple):
+                continue
+            list_of_routes_which_needs_schema.append(route)
+        return list_of_routes_which_needs_schema
```

Comment:
Optional: you can replace all this by `[return route for route in main.URLS if not isinstance(route, tuple)]`

(Though your TODO comment might need to be updated accordingly if you do that.)